lp-collaborative-workspace/lp-cw-bridge/lp-cw-bridge-transformer/src/main/java/transformation/controller/ATLTransformation.java

```diff
@@ -0,0 +1,182 @@
+package it.univaq.disim.transformation.controller;
+
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.xmi.impl.EcoreResourceFactoryImpl;
+import org.eclipse.m2m.atl.common.ATLExecutionException;
+import org.eclipse.m2m.atl.core.ATLCoreException;
+import org.eclipse.m2m.atl.core.IExtractor;
+import org.eclipse.m2m.atl.core.IInjector;
+import org.eclipse.m2m.atl.core.IModel;
+import org.eclipse.m2m.atl.core.IReferenceModel;
+import org.eclipse.m2m.atl.core.ModelFactory;
+import org.eclipse.m2m.atl.core.emf.EMFExtractor;
+import org.eclipse.m2m.atl.core.emf.EMFInjector;
+import org.eclipse.m2m.atl.core.emf.EMFModelFactory;
+import org.eclipse.m2m.atl.core.launch.ILauncher;
+import org.eclipse.m2m.atl.engine.compiler.atl2006.Atl2006Compiler;
+import org.eclipse.m2m.atl.engine.emfvm.launch.EMFVMLauncher;
+
+
+public class ATLTransformation {
+	private IModel inModel;
+	private IModel outModel;
+
+	private IReferenceModel inmodelMetamodel;
+	private IReferenceModel outmodelMetamodel;
+
+	private String modules; //Con questo intendiamo il file della trasformazione .atl
+
+	private String inTag;
+	private String outTag;
+
+	private Map<String, Object> options;
+
+	public ATLTransformation() throws IOException {
+		options = new HashMap<String, Object>();
+		options.put("supportUML2Stereotypes", "false");
+		options.put("printExecutionTime", "true");
+		options.put("OPTION_CONTENT_TYPE", "false");
+		options.put("allowInterModelReferences", "false");
+		options.put("step", "false");
+		Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().put("ecore", new EcoreResourceFactoryImpl());
+
+	}
+	
+	
+	
+	/**
+	 * The function is the only one to be visible to the outside and is the one that initiates the whole process of transformation.
+	 * @param String model_in, String metamodel_in, String metamodel_out, String modules, String inTag, String outTag, String outPath

```

Comment:
I will provide comments that you request as soon as possible! I also try to give an example of how the function works.